cpp/src/arrow/type.cc

```diff
@@ -2462,6 +2462,10 @@ void InitStaticData() {
   // Temporal types
   g_temporal_types = {date32(),
                       date64(),
+                      duration(TimeUnit::SECOND),
+                      duration(TimeUnit::MILLI),
+                      duration(TimeUnit::MICRO),
+                      duration(TimeUnit::NANO),
```

Comment:
I'm not sure that's right. Given that we have a separate category here for interval types, it probably means we also want a dedicated category for durations?

Reply:
That may be the case. I opted to include them in the temporal group based on their common usage of `TimeUnit`s and an integral `c_type`, but perhaps that's an arbitrary categorization. On first glance, it seemed like an unintentional omission.

Now that you mention it, the `kernels/scalar_cast_string.cc` implementation must exclude interval types as well, since it only sources from `TemporalTypes` in `AddTemporalToStringCasts` - so maybe its a better idea to solve the issue there (after adding a separate category for duration types, like you suggested).

Reply:
Yes, it would be better to solve the issue there IMHO.